### PR TITLE
Subclass route options

### PR DESCRIPTION
**Pull request by bsudekum**
This creates a sublcass of `RouteOptions` called `NavigationRouteOptions`. 

/cc @frederoni @1ec5 @ericrwolfe

### CODE REVIEW

File: MapboxCoreNavigation/NavigationRouteOptions.swift [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import Foundation
+import MapboxDirections
+
+/**
+ A `NavigationRouteOptions` object is a structure that specifies the criteria for results returned by the Mapbox Directions API.
+ 
+ `NavigationRouteOptions` is a subclass of `RouteOptions` that has been optimized for navigation. Pass an instance of this class into the `Directions.calculate(_:completionHandler:)` method.
```
Comment (1ec5) [inline MapboxCoreNavigation/NavigationRouteOptions.swift]: This is kind of silly, but what if we extended Directions to make NavigationRouteOptions a little more discoverable to folks who don’t read documentation?

```swift
extension Directions {
    @objc(calculateDirectionsWithNavigationOptions:completionHandler:)
    open func calculate(_ options: NavigationOptions, completionHandler: @escaping CompletionHandler) -> URLSessionDataTask {
        return calculate(options as RouteOptions, completionHandler)
    }
}
```

Disclaimer: I haven’t tried this; just playing around in a text box in a browser.

File: MapboxCoreNavigation/NavigationRouteOptions.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ */
+@objc(MBNavigationRouteOptions)
+open class NavigationRouteOptions: RouteOptions {
+    
+    /**
+     Initializes a navigation route options object for routes between the given waypoints and an optional profile identifier optimized for navigation.
+     
+     - parameter waypoints: An array of `Waypoint` objects representing locations that the route should visit in chronological order. The array should contain at least two waypoints (the source and destination) and at most 25 waypoints. (Some profiles, such as `MBDirectionsProfileIdentifierAutomobileAvoidingTraffic`, [may have lower limits](https://www.mapbox.com/api-documentation/#directions).)
+     - parameter profileIdentifier: A string specifying the primary mode of transportation for the routes. This parameter, if set, should be set to `MBDirectionsProfileIdentifierAutomobile`, `MBDirectionsProfileIdentifierAutomobileAvoidingTraffic`, `MBDirectionsProfileIdentifierCycling`, or `MBDirectionsProfileIdentifierWalking`. `MBDirectionsProfileIdentifierAutomobile` is used by default.
```
Comment (1ec5) [inline MapboxCoreNavigation/NavigationRouteOptions.swift]: I’m a little uncomfortable with mirroring all the RouteOptions documentation here. Could we get away with a `- see:` command? We might need to beef up [this line](https://github.com/mapbox/mapbox-navigation-ios/blob/5f3856376eaead146a58688bbed40df842b54e9d/scripts/document.sh#L50) that inserts links from the MapboxNavigation docset to the MapboxDirections docset.

File: MapboxCoreNavigation/NavigationRouteOptions.swift [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+     Initializes a route options object for routes between the given geographic coordinates and an optional profile identifier optimized for navigation.
+     
+     - parameter coordinates: An array of geographic coordinates representing locations that the route should visit in chronological order. The array should contain at least two locations (the source and destination) and at most 25 locations. Each coordinate is converted into a `Waypoint` object.
+     - parameter profileIdentifier: A string specifying the primary mode of transportation for the routes. This parameter, if set, should be set to `MBDirectionsProfileIdentifierAutomobile`, `MBDirectionsProfileIdentifierAutomobileAvoidingTraffic`, `MBDirectionsProfileIdentifierCycling`, or `MBDirectionsProfileIdentifierWalking`. `MBDirectionsProfileIdentifierAutomobile` is used by default.
+     */
+    public convenience init(coordinates: [CLLocationCoordinate2D], profileIdentifier: MBDirectionsProfileIdentifier? = .automobileAvoidingTraffic) {
+        self.init(waypoints: coordinates.map { Waypoint(coordinate: $0) }, profileIdentifier: profileIdentifier)
+    }
+    
+    public required init?(coder decoder: NSCoder) {
```
Comment (1ec5) [inline MapboxCoreNavigation/NavigationRouteOptions.swift]: Nit: it probably isn’t necessary to override this method at this time.

File: MapboxCoreNavigation/NavigationRouteOptions.swift [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import Foundation
+import MapboxDirections
+
+/**
+ A `NavigationRouteOptions` object is a structure that specifies the criteria for results returned by the Mapbox Directions API.
```
Comment (1ec5) [inline MapboxCoreNavigation/NavigationRouteOptions.swift]: We should briefly distinguish between this class and RouteOptions in the first sentence, for the benefit of developers looking at the code completion popup:

> A `NavigationRouteOptions` object specifies turn-by-turn-optimized criteria for results returned by the Mapbox Directions API.

File: MapboxCoreNavigation/NavigationRouteOptions.swift [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+ 
+ `NavigationRouteOptions` is a subclass of `RouteOptions` that has been optimized for navigation. Pass an instance of this class into the `Directions.calculate(_:completionHandler:)` method.
+ */
+@objc(MBNavigationRouteOptions)
+open class NavigationRouteOptions: RouteOptions {
```
Comment (1ec5) [inline MapboxCoreNavigation/NavigationRouteOptions.swift]: I had been thinking of calling this class NavigationOptions, but NavigationRouteOptions makes more sense. 👍

File: MapboxCoreNavigation/RouteController.swift [hunk 6]
```diff
@@ -182,6 +182,10 @@ open class RouteController: NSObject {
         self.resumeNotifications()
         self.startEvents(route: route)
         self.resetSession()
+        
+        if !(route.routeOptions is NavigationRouteOptions) {
+            print("`Route` was created using `RouteOptions` and not `NavigationRouteOptions`. Although not required, this may lead to a suboptimal navigation experience.")
```
Comment (1ec5) [inline MapboxCoreNavigation/RouteController.swift]: Let’s spell out what the developer gains by using NavigationRouteOptions: congestion coloring along the route line, better ETAs, what else?

File: MapboxCoreNavigation/RouteOptions.swift [hunk 7]
```diff
@@ -1,45 +1,13 @@
...
 import MapboxDirections
 
 extension RouteOptions {
     var activityType: CLActivityType {
         switch self.profileIdentifier {
-        case MBDirectionsProfileIdentifier.cycling, MBDirectionsProfileIdentifier.walking:
-            return .fitness
-        default:
-            return .automotiveNavigation
+            case MBDirectionsProfileIdentifier.cycling, MBDirectionsProfileIdentifier.walking:
```
Comment (1ec5) [inline MapboxCoreNavigation/RouteOptions.swift]: Is Xcode 9 indenting `case` now? 🙀

File: MapboxCoreNavigation/NavigationRouteOptions.swift (hunk 3 above)
Comment (bsudekum) [inline MapboxCoreNavigation/NavigationRouteOptions.swift]: It suggests a fix-it here.

File: MapboxCoreNavigation/NavigationRouteOptions.swift (hunk 2 above)
Comment (bsudekum) [inline MapboxCoreNavigation/NavigationRouteOptions.swift]: What would the `- see` command look like here?

Comment (1ec5) [reply MapboxCoreNavigation/NavigationRouteOptions.swift]: Sorry, that should be [`- SeeAlso:`](https://developer.apple.com/library/content/documentation/Xcode/Reference/xcode_markup_formatting_ref/SeeAlso.html).

File: MapboxCoreNavigation/NavigationRouteOptions.swift [hunk 8]
```diff
@@ -0,0 +1,52 @@
+import Foundation
+import MapboxDirections
+
+/**
+ A NavigationRouteOptions object specifies turn-by-turn-optimized criteria for results returned by the Mapbox Directions API.
```
Comment (1ec5) [inline MapboxCoreNavigation/NavigationRouteOptions.swift]: Put `NavigationRouteOptions` in backticks for consistency.

File: MapboxCoreNavigation/RouteController.swift [hunk 9]
```diff
@@ -185,6 +185,10 @@ open class RouteController: NSObject {
         self.resumeNotifications()
         self.startEvents(route: route)
         self.resetSession()
+        
+        if !(route.routeOptions is NavigationRouteOptions) {
+            print("`Route` was created using `RouteOptions` and not `NavigationRouteOptions`. Although not required, this may lead to a suboptimal navigation experience. Without `NavigationRouteOptions`, it is not guaranteed you will get congestion along the route line, better ETAs and ETA label color dependent on congestion.")
```
Comment (1ec5) [inline MapboxCoreNavigation/RouteController.swift]: None of these features are implemented in Core Navigation, only in MapboxNavigation. Can we move this check to NavigationViewController’s initializer?

Also, use a (lazily initialized) global or static property to track whether this message has been printed before and avoid printing it a second time.

**Pull request merged**